src/visualization.jl

```diff
@@ -23,16 +32,105 @@ function Graph(p::AbstractPetriNet)
   stmts = vcat(statenodes, transnodes)
 
   edges = map(1:nt(p)) do k
-    k_name = tname(p, k)
-    vcat(edgify(countmap(map(x->sname(p,x), inputs(p, k))), k_name, false),
-         edgify(countmap(map(x->sname(p, x), outputs(p, k))), k_name, true))
+    vcat(edgify(countmap(Vector{Any}(inputs(p, k))), k, false),
+         edgify(countmap(Vector{Any}(outputs(p, k))), k, true))
   end |> flatten |> collect
 
   stmts = vcat(stmts, edges)
   g = Graphviz.Digraph("G", stmts; graph_attrs=graph_attrs, node_attrs=node_attrs, edge_attrs=edge_attrs)
   return g
 end
 
+function Subgraph(p::AbstractPetriNet; label="cluster", pre="")
+   statenodes = [Node("\"$(pre)s$s\"",
+                      Attributes(:label => "$(sname(p, s))", :shape=>"circle",
+                                 :color=>"#6C9AC3")) for s in 1:ns(p)]
+
+  transnodes = [Node("\"$(pre)t$k\"",
+                     Attributes(:label => "$(tname(p, k))", :shape=>"square",
+                                :color=>"#E28F41")) for k in 1:nt(p)]
+
+  node_attrs  = Attributes(:shape=>"plain", :style=>"filled", :color=>"white")
+  edge_attrs  = Attributes(:splines=>"splines")
+
+  stmts = vcat(statenodes, transnodes)
+
+  edges = map(1:nt(p)) do k
+    vcat(edgify(countmap(Vector{Any}(inputs(p, k))), k, false; pre=pre),
```

Comment:
I don't like this cast to Vector{Any}. It looks like it is solving a problem with countmap on empty collections. Can we define another method on countmap to catch this bug instead of casting all our vectors to vector of any here?